local_planner/src/nodes/waypoint_generator.cpp

```diff
@@ -165,13 +165,11 @@ void WaypointGenerator::reachGoalAltitudeFirst() {
   // goto_position is a unit vector pointing straight up/down from current
   // location
   output_.goto_position = position_;
-  goal_.x() = position_.x();  // Needed so adaptSpeed can clamp to goal
-  goal_.y() = position_.y();
 
   // Only move the setpoint if drone is in the air
   if (is_airborne_) {
     // Ascend/Descend to goal altitude
-    if (position_.z() <= goal_.z()) {
+    if (position_.z() <= planner_info_.starting_height) {
```

Comment:
I'm a bit concerned what the `else` here might do. I could imagine the drone just passing the starting height (which is half the goal height), then the goto_position will point straight down until the firmware has updated the next waypoint and thus gotten the planner to switch out of this state...

Reply:
I am not sure I follow

Reply:
Let's say the drone is taking off, `goal_.z` is 4m. Once position_.z is >2.0 m (= starting_height), the goto_position will make the drone descend again, even though we still want to rise

Reply:
If the starting_height has been reached it won't go trought the `reachGoalAltitudeFirst()` because of https://github.com/PX4/avoidance/pull/249/files#diff-2be3b67376d56938a587e393161de500R177

Reply:
Fair enough